lib/core/presentation/widgets/poap/poap_claim_transfer_controller_widget/poap_claim_transfer_controller_widget.dart

```diff
@@ -0,0 +1,93 @@
+import 'package:app/core/application/auth/auth_bloc.dart';
+import 'package:app/core/application/poap/poap_claim_transfer_subscription_bloc/poap_claim_transfer_subscription_bloc.dart';
+import 'package:app/core/domain/poap/poap_enums.dart';
+import 'package:app/core/presentation/widgets/poap/poap_claimed_poup/poap_claimed_popup.dart';
+import 'package:app/i18n/i18n.g.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class PoapClaimTransferControllerWidget extends StatelessWidget {
+  const PoapClaimTransferControllerWidget({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<AuthBloc, AuthState>(
+      builder: (context, state) => state.maybeWhen(
+        orElse: () => const SizedBox.shrink(),
+        authenticated: (_) => BlocProvider(
+          create: (context) => PoapClaimTransferSubscriptionBloc()
+            ..add(PoapClaimTransferSubscriptionEvent.start()),
+          child: const _PoapClaimTransferControllerWidgetView(),
+        ),
+      ),
+    );
+  }
+}
+
+class _PoapClaimTransferControllerWidgetView extends StatefulWidget {
+  const _PoapClaimTransferControllerWidgetView();
+
```

Comment:
There is no point of separate body of main Widget into a private widget here. It hard to debug and testing in the future. Please merge it into one.

Reply:
I'm fine with this, because think in other side that this will make consistent structure + good for scale when this file growth more

Reply:
Then make the sub widget into stateless, no point of making it stateful

Reply:
Yep, make sense 👍

Reply:
the reason for this is that we cannot use bloc provider and blocbuilder(or bloc listener) in the same level of widget, thats why split it into another subview to make use or blocbuilder and bloclistener
@roncoderpush @sieuhuflit

Reply:
I see 👍